enhancements/network/bgp-overview.md

```diff
@@ -0,0 +1,152 @@
+Calico uses BGP.
+
+There are other ways this control plane functionality can be implemented, such
+as using the Kubernetes API or by using some other custom control plane
+technology.  Both OpenShift-SDN and OVN-Kubernetes use alternative methods to
+implement their control planes and it is not feasible to change them to use
+BGP for this purpose.
+
+### External Service Load Balancing
+
+Kubernetes Services with `type=LoadBalancer` are typically used as an
+abstraction in front of a cloud's load balancer service.  Many on premise
+cluster environments do not have a load balancer service available, so these
+Services may not be used.
+
+It is possible to implement `LoadBalancer` Services for these on premise
+clusters other ways.  One way involves using a BGP speaker on Nodes to publish
+routes to Nodes where `LoadBalancer` IP addresses may be reached.  [MetalLB][2]
+is one project that implements this technique.  [OpenShift Enhancement #356][3]
+discusses the use of MetalLB with OpenShift in more detail.
+
+### Exposing Pods or Services Directly
+
+Some users have expressed an interest in publishing routes for the Pod and
+Service networks via BGP.  This bypasses all of the usual interfaces used for
+getting traffic to applications in a cluster, so it's a bit of an awkward
+capability to offer from an architectural perspective.
+
+If we wanted to offer this, it would not be very difficult.  At least with
+OpenShift-SDN and OVN-Kubernetes, Nodes are already set up to handle traffic
+destined for Pod or Service cluster IP addresses if traffic were to arrive
+destined for those addresses.
+
+To enable this using BGP, a BGP speaker must run on each Node where we would
+like to serve as a router into the cluster network.  Each Node can publish a
+route for the Pod and Service network and upstream routers may use ECMP to
+distribute traffic among the Nodes.
+
+A future enhancement needs to be filed to pursue this integration in more
+detail.
+
+### Virtual Network Interconnect
+
+The Pod network is by default an isolated virtual network.  There is sometimes
+a desire to interconnect multiple virtual networks together.  This could be
+between multiple Kubernetes clusters, but it doesn't have to be the case.
+
+BGP EVPN (Ethernet VPN, or Virtual Private Network) is a BGP technology that is
+discussed in the context of this use case.  BGP EVPN can be used to advertise
+MAC addresses and IP-MAC address bindings.  BGP EVPN is not a full solution to
+this use case by itself, as it is a technology that can help implement part of
+the control plane.  There are more details that must be explored, including
+what the data plane looks like between the two clusters.
+
+This is a significantly non-trivial effort.  Future enhancements are needed to
+discuss this in much more detail.
+
+## Common Design Considerations
+
+This section summarizes the common design considerations for OpenShift in
+support of BGP related features.
```

Comment:
As mentioned above, in the past we believed that some customers would not want OpenShift to be modifying the routing of their networks (and thus we can't have any mandatory usage of BGP). Assuming that is still the case, there should be a point about that here.

Reply:
Excellent point!  I'll add this